src/MCPClient/lib/clientScripts/move_to_backlog.py

```diff
@@ -191,11 +190,7 @@ def main(job, transfer_id, transfer_path, created_at):
     _index_transfer(job, transfer_id, transfer_path)
 
     logger.info("Calculating size...")
-    size = 0
-    for dirpath, _, filenames in scandir.walk(transfer_path):
-        for filename in filenames:
-            file_path = os.path.join(dirpath, filename)
-            size += os.path.getsize(file_path)
+    size = get_dir_size(transfer_path)
```

Comment:
These are great observations! I agree that we should avoid doing the same thing multiple places, and that we should be passing this information along rather than re-calculating it later. I also lean toward amending the indexing function signature. I'll look into amending what's there to fit that model, with an eye toward making sure whatever changes are made also play nicely with our re-indexing scripts for existing transfers in the backlog.

Reply:
Hi Ross! I made the changes you suggested in commit 3fff680. `get_dir_size` helper function has been refactored as `get_bag_size` and accepts a bag object and the path to the bag as arguments (the latter in case `Payload-Oxum` as an optional attribute is not present in bag-info.txt and we have to calculate size with a dirwalk)